src/components/rule.js

```diff
@@ -1,30 +1,26 @@
-var _ = require('lodash');
-
-var Rule = function(data, id){
-	this.id = id;
-	this.isDirectional = data.isDirectional;
-
-	this.cause = data.cause;
-	this.consequent = data.consequent;
-
-	this.consequentThing = data.consequentThing;
-	this.mutations = data.mutations;
-}
-
-Rule.prototype.getSource = function(){
-	return this.cause.type[0];
-}
-
-Rule.prototype.getTarget = function(){
-	return this.cause.type[2];
-}
-
-Rule.prototype.getConsequentTarget = function(){
-	return this.consequent.type[2];
+'use strict';
+
+module.exports = class Rule {
+  constructor(data, id) {
+  	this.id = id;
+  	this.isDirectional = data.isDirectional;
+
+  	this.cause = data.cause;
+  	this.consequent = data.consequent;
+
+  	this.consequentThing = data.consequentThing;
+  	this.mutations = data.mutations;
+  }
+  getSource() {
+  	return this.cause.type[0];
+  }
+  getTarget() {
+  	return this.cause.type[2];
+  }
+  getConsequentTarget() {
+  	return this.consequent.type[2];
+  }
+  getActionType() {
+  	return this.cause.type[1];
+  }
```

Comment:
I rewrote this while I was in there to demonstrate how this would look utilizing classes. I'm happy to refactor like this as I go or to leave it as is, just wanted to give the option 😁